setup.sh

```diff
@@ -674,12 +674,15 @@ echo "#!/bin/sh" >$config_dir/.reload-init
 echo "echo Reloading Webmin server in $wadir" >>$config_dir/.reload-init
 echo "pidfile=\`grep \"^pidfile=\" $config_dir/miniserv.conf | sed -e 's/pidfile=//g'\`" >>$config_dir/.reload-init
 echo "kill -USR1 \`cat \$pidfile\`" >>$config_dir/.reload-init
+# Switch to systemd from init (intermediate)
 if [ "$killmodenonesh" = "1" ] && [ -x "$systemctlcmd" ]; then
 	current_version=`cat "$config_dir/version" 2>/dev/null`
 	ancient_version=`echo $current_version 1.994 | awk '{if ($1 < $2) print 1; else print 0}'`
 	if [ "$ancient_version" = "1" ]; then
-		echo "$config_dir/.stop-init" >>$config_dir/.reload-init
-		echo "$config_dir/start" >>$config_dir/.reload-init
+		echo "#!/bin/sh" >$config_dir/.reload-init-systemd
+		echo "$config_dir/.stop-init" >>$config_dir/.reload-init-systemd
```

Comment:
Doesn't this need to be `.stop-init --kill` so that it works if the PID file is missing?

Reply:
Oh, no no! If we use `--kill` it will take down the whole process. It will be handled correctly either way for these rare cases, i.e. a user won't see a breakage at all. At least, great majority of users won't see an actual breakage at all. And I highly doubt that anyone who had Webmin actually finished to upgrade (files placed on the disk) and using Authentic Theme will ever see a breakage, even in case Webmin wasn't restarted correctly.

Transition from `init` to `systemd` is complicated because we don't know how older versions of Webmin (before 1.994) were started (using older `/etc/webmin/start` script or using `systemctl` command). In addition there is also to consider this odd situation with missing pid file.

We simply cannot and must not use `--kill`.

Reply:
What about the case where the PID file is missing though? Isn't --kill needed to handle that, because `.stop-init` contains : 

```
  if [ "$1" = "--kill" ]; then
    (ps axf | grep "webmin\/miniserv\.pl" | awk '{print "kill -9 -- -" $1}' | bash) 2>/dev/null
  fi
```

Reply:
This is meant to be only called from a console. We cannot call it from the process itself. Webmin like no other, restarts itself and this is where there great complexity is coming from.